### ISSUE TITLE
Timeseries and NodeGraph container inconsistent when together

### ISSUE THREAD

**Issue by harisrozajac** [grafana/grafana#75711]
_Originally posted by @gelicia in https://github.com/grafana/grafana/pull/74536#pullrequestreview-1647555370_

How to reproduce:

1. Use test datasource
2. First query is random walk
3. Add query
4. Make second query node graph

Show/hide toggling the first query and second query shows inconsistent results. Toggling one query shouldn't affect the display of the other query, but in this case when Timeseries is toggled NodeGraph disappears and vice versa. 

See video:
https://github.com/grafana/grafana/assets/58232930/160dd5a9-0511-48fd-ac09-877e9d61c5cd

**Comment by ThinhNguyenHoang1** [comment]
Hi, @harisrozajac I want to tackle this issues. Can you assign it to me ? I want this to be my first OSS contribution to grafana.

**Comment by ThinhNguyenHoang1** [comment]
Just to clarify so that I am sure I understand the problem and expectation correctly. 

The current behavior is that when you toggle one of the queries, all the queries are rerun and their display panels are redrew (1). But for the scenarios mentioned, the result display panels are inconsistent - missing some result panels (2) 

So, The expected behavior we want is that when you toggling the query, this is only an UI filtering ops they shouldn't cause a rerun of all queries (1) and no result panels should be missing (2).

Am I understanding this correctly ? 
@harisrozajac

**Comment by gelicia** [comment]
@ThinhNguyenHoang1 Right. When you run multiple queries with different visualizations in explore, toggling the queries to show/hide should only show/hide the visualization for that query.

**Comment by ThinhNguyenHoang1** [comment]
Hello, after inspecting the code I found that the root cause of inconsistency is that the **NodeGraph Scenario's query overrides the data of other queries as in here.** 

https://github.com/grafana/grafana/blob/08803297968350bb1ca78e9990094c9049d47e04/public/app/features/explore/state/query.ts#L1306

Note: I checked the that the code calculate correct data for each type and called its handler properly. But when choosing multiple Node Graph Query  only 1 will be visualized (this is also the behavior with FlameGraph). As I am quite new to grafana, I am not sure if this is intended ? 

Any input or suggestion on which direction should we take is welcome!!

**Comment by ThinhNguyenHoang1** [comment]
Hello, after inspecting the code, I found that the issues with consistency resolve around these two snippet of code:

The sequential handling of queries' response
https://github.com/grafana/grafana/blob/08803297968350bb1ca78e9990094c9049d47e04/public/app/features/explore/state/query.ts#L654


which sequentially apply the reducer logic here for each query response: 
https://github.com/grafana/grafana/blob/08803297968350bb1ca78e9990094c9049d47e04/public/app/features/explore/state/query.ts#L1306

The inconsistency stem from the fact that NodeGraph query response will override the response data other queries need to visualize and vice versa. Causing the missing of some visualization when NodeGraph query is chosen.


To continue forward, I want to confirm about which action we should take:
+ The current logic will visualize only 1 NodeGraph query when N NodeGraph query is configured (This is NOT the case with FlameGraph and RandomWalk). Is this intended ? What is expected behavior for this ?
